util/src/openssl_utils.cpp

```diff
@@ -0,0 +1,258 @@
+// Concord
+//
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+
+#include "openssl_utils.hpp"
+#include "Logger.hpp"
+#include "assertUtils.hpp"
+#include "hex_tools.h"
+#include "openssl_crypto.hpp"
+#include "crypto/eddsa/EdDSA.h"
+
+#include <regex>
+#include <utility>
+
+namespace concord::crypto::openssl {
+
+using concord::util::openssl_utils::UniquePKEY;
+using concord::util::openssl_utils::UniqueOpenSSLPKEYContext;
+using concord::util::openssl_utils::UniqueOpenSSLX509;
+using concord::util::openssl_utils::UniqueOpenSSLBIO;
+
+string CertificateUtils::generateSelfSignedCert(const string& origin_cert_path,
+                                                const string& public_key,
+                                                const string& signing_key) {
+  auto deleter = [](FILE* fp) {
```

Comment:
use std file streams instead

Reply:
We need to have `FILE*` instead of file streams because `PEM_read_X509` wants the `FILE*` as the first param.